util/keyfile/src/lib.rs

```diff
@@ -107,90 +122,102 @@ fn to_io_error<E: 'static + std::error::Error + Send + Sync>(err: E) -> std::io:
     std::io::Error::new(std::io::ErrorKind::Other, Box::new(err))
 }
 
-fn prost_to_io_error(err: mc_util_serial::DecodeError) -> std::io::Error {
-    std::io::Error::new(std::io::ErrorKind::Other, Box::new(ProstError { err }))
-}
-
-// Helper boilerplate mapping prost::DecodeError to io::Error
-#[derive(Debug)]
-struct ProstError {
-    pub err: mc_util_serial::DecodeError,
-}
+#[cfg(test)]
+mod testing {
 
-impl std::fmt::Display for ProstError {
-    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
-        write!(fmt, "prost deserialization failed: {}", self.err)
-    }
-}
+    use super::*;
+    use bip39::{Language, MnemonicType};
+    use mc_account_keys::AccountKey;
+    use mc_account_keys_slip10::{Slip10Key, Slip10KeyGenerator};
 
-impl std::error::Error for ProstError {
-    fn description(&self) -> &str {
-        "prost deserialization failed"
+    /// Test that round-tripping through a keyfile without fog gets the same
+    /// result as creating the key directly.
+    #[test]
+    fn keyfile_roundtrip_no_fog() {
+        let dir = tempfile::tempdir().expect("Could not create temp dir");
```

Comment:
This isn't for this PR, but we should make a test utils crate that uses `TempDir::new_in()` to put all these into the `OUT_DIR`, and replace all our localized `tempfile` users.

Reply:
Created a ticket for this.